src/emailaction.cpp

```diff
@@ -875,7 +882,28 @@ EasInvitationResponse::~EasInvitationResponse()
 
 void EasInvitationResponse::execute()
 {
-    _protocolAction->protocolRequest(_accountId, "meetingresponse", _responseData);
+    QString responseString;
+    switch (_response) {
+    case EmailAgent::InvitationResponseAccept:
+        responseString = "accept";
+        break;
+    case EmailAgent::InvitationResponseTentative:
+        responseString = "tentative";
+        break;
+    case EmailAgent::InvitationResponseDecline:
+        responseString = "decline";
+        break;
+    default:
+        qCDebug(lcEmail) << "EAS: Invalid calendar response specified";
+        return;
+    }
+
+    QVariantMap data;
+    data.insert("messageId", _messageId.toULongLong());
+    data.insert("response", responseString);
+    data.insert("replyMessageId", _replyMessageId.toULongLong());
+
+    _protocolAction->protocolRequest(_accountId, "meetingresponse", data);
```

Comment:
It's cleaner to pass the id's and only create the QVariantMap as late as possible - after everything before it has succeeded - since it's technically a protocol implement detail. Looks good!